comp/core/agenttelemetry/agenttelemetryimpl/config.go

```diff
@@ -0,0 +1,418 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package agenttelemetryimpl
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultSheduleStartAfter = 30  // 30 seconds
+	defaultShedulePeriod     = 900 // 15 minutes
+)
+
+// Config is the top-level config for agent telemetry
+type Config struct {
+	Enabled  bool      `yaml:"enabled"`
+	Profiles []Profile `yaml:"profiles"`
+
+	// compiled
+	schedule map[Schedule][]*Profile
+}
+
+// Profile is a single agent telemetry profile
+type Profile struct {
+	// parsed
+	Name     string             `yaml:"name"`
+	Metric   *AgentMetricConfig `yaml:"metric,omitempty"`
+	Status   *AgentStatusConfig `yaml:"status"`
+	Schedule *Schedule          `yaml:"schedule"`
+
+	// compiled
+	statusExtraBuilder []jBuilder
+	metricsMap         map[string]*MetricConfig
+	excludeZeroMetric  bool
+	excludeTags        map[string]any
```

Comment:
It seems we only assign `struct{}{}` here, would this work?
```suggestion
	excludeTags        map[string]struct{}
```

Reply:
> It seems we only assign struct{}{} here, would this work?

No, actually. Because excluded tag can be specified with or without value. And if there is a value I change the node to another map. See https://github.com/DataDog/datadog-agent/pull/22771/files#diff-4d02442e857d5db8627ad2e0fc6f7cccc9dfff7e0765db5448878d3cfc817935R282-R284. Before `any` I did have `interface{}` type before.